spec/unit/provider/nameservice_spec.rb

```diff
@@ -301,4 +301,42 @@
     end
   end
 
+  describe "comments_in_sync?(is,should)" do
+    it "should force_encoding of should-value to binary if is-value is binary" do
+      is_value = "foo".force_encoding(Encoding::ASCII_8BIT)
+      should_value = "bar".force_encoding(Encoding::UTF_8)
+
+      expect(should_value.encoding).to be(Encoding::UTF_8)
+      provider.comments_insync?(is_value, should_value)
+      expect(should_value.encoding).to be(Encoding::ASCII_8BIT)
+    end
+
+    it "should not modify encoding of should-value if is-value is not in binary" do
+      is_value = "foo".force_encoding(Encoding::ASCII)
+      should_value = "bar".force_encoding(Encoding::UTF_8)
+
+      expect(should_value.encoding).to be(Encoding::UTF_8)
+      provider.comments_insync?(is_value, should_value)
+      expect(should_value.encoding).to be(Encoding::UTF_8)
+    end
+
+    it "should do basic string comparison of the is-value with the should-value" do
+      is_value = "foo"
+      should_value = "bar"
+      is_value.expects(:'==').with("bar")
+      provider.comments_insync?(is_value, should_value)
+    end
+
+    it "should return false if the is-value and should-value are not equal" do
```

Comment:
Should there be a test for returning false for different strings with different encodings?